app/src/main/java/com/example/mona/facebookoffline/MainActivity.java

```diff
@@ -2,21 +2,57 @@
 
 import android.app.Activity;
 
+import android.content.Intent;
 import android.os.Bundle;
 import android.view.View;
 import android.widget.Button;
+import android.widget.Toast;
 
 import com.example.mona.facebookoffline.auth.LoginFragment;
+import com.facebook.AccessToken;
+import com.facebook.FacebookException;
+import com.facebook.FacebookSdk;
+import com.facebook.login.LoginResult;
 
 /**
  * Main activity, allows a user to login, create new posts, and access old posts
  */
-public class MainActivity extends Activity {
+public class MainActivity extends Activity implements LoginFragment.LoginListener {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        // Initialize FB sdk
+        FacebookSdk.sdkInitialize(this);
```

Comment:
Now that this page also takes care of login and is always shown, I don't think it's necessary to move this line from LoginFragment to here. That said, both solutions are kind of suboptimal since we have to keep moving the line based on what Activity is launched first. There's a better solution to this by overriding the Application class, but we can deal with that later! Just wanted to point it out